mirgecom/fluid.py

```diff
@@ -391,10 +391,14 @@ def velocity_gradient(discr, cv, grad_cv):
         \partial_{x}\mathbf{v}_{y}&\partial_{y}\mathbf{v}_{y} \end{array} \right)$
 
     """
-    velocity = cv.momentum / cv.mass
-    return (1/cv.mass)*make_obj_array([grad_cv.momentum[i]
-                                       - velocity[i]*grad_cv.mass
+    vel = cv.velocity
+    obj_ary = (1/cv.mass)*make_obj_array([grad_cv.momentum[i]
+                                       - vel[i]*grad_cv.mass
                                        for i in range(cv.dim)])
+    grad_v = np.empty(shape=(cv.dim, cv.dim), dtype=object)
+    for idx, v in enumerate(obj_ary):
+        grad_v[idx] = v
+    return grad_v
```

Comment:
Can you explain why you don't just simply return `obj_ary` and instead make an entirely new object array with the same values of `obj_ary`?

Reply:
Because I need the returned structure to have the 2D shape, not flat. Let me see if I can do it in a less awkward way.

Reply:
See how this grabs ya (e2d4552).

Reply:
Resolving this @thomasgibson due to it now being fully replaced by @majosm suggestion.